plugins/modules/nd_multi_cluster_connectivity.py

```diff
@@ -0,0 +1,354 @@
+
+#     return payload
+
+
+def create_cluster(nd):
+    path = "/api/v1/infra/clusters"
+    response = nd.proposed
+    if not nd.module.check_mode:
+        response = nd.request(path, method="POST", data=nd.proposed)
+    return response
+
+
+def update_cluster(nd):
+    path = "/api/v1/infra/clusters"
+    fabric_name_for_api_path = nd.existing.get("spec", {}).get("name")
+    response = payload = nd.proposed
+    payload["spec"]["name"] = fabric_name_for_api_path
+    if not nd.module.check_mode:
+        response = nd.request("{0}/{1}".format(path, fabric_name_for_api_path), method="PUT", data=payload)
+    return response
+
+
+def delete_cluster(nd):
+    path = "/api/v1/infra/clusters"
+    fabric_name_for_api_path = nd.existing.get("spec", {}).get("name")
+    response = payload = {}
+    if nd.existing.get("spec", {}).get("clusterType", "").upper() == "APIC":
+        payload = {
+            "credentials": {
+                "user": nd.proposed.get("spec", {}).get("credentials", {}).get("user"),
+                "password": nd.proposed.get("spec", {}).get("credentials", {}).get("password"),
+            }
+        }
+    if not nd.module.check_mode:
+        response = nd.request("{0}/{1}/remove".format(path, fabric_name_for_api_path), method="POST", data=payload)
+    return response
+
+
+def main():
+    argument_spec = nd_argument_spec()
+    argument_spec.update(
+        config=dict(
+            type="list",
+            elements="dict",
+            options=dict(
+                cluster_type=dict(type="str", choices=["nd", "apic"]),
+                cluster_hostname=dict(type="str"),
+                cluster_username=dict(type="str"),
+                cluster_password=dict(type="str", no_log=True),
+                cluster_login_domain=dict(type="str"),
+                multi_cluster_login_domain=dict(type="str"),
+                fabric_name=dict(type="str"),
+                license_tier=dict(type="str", choices=["advantage", "essentials", "premier"]),
+                features=dict(type="list", elements="str", choices=["telemetry", "orchestration"]),
+                inband_epg=dict(type="str"),
+                security_domain=dict(type="str"),
+                validate_peer_certificate=dict(type="bool"),
+            ),
+            required=True,
+        ),
+        state=dict(type="str", default="merged", choices=["merged", "replaced", "overridden", "deleted"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+    )
+
+    nd = NDModule(module)
+    desired_clusters_from_module_input = nd.params.get("config")
+    state = nd.params.get("state")
+
+    desired_clusters_map = {}
+    for cluster_data_input in desired_clusters_from_module_input:
+        normalized_payload = module_input_to_api_payload(cluster_data_input)
+        desired_clusters_map[get_cluster_unique_key(normalized_payload)] = normalized_payload
```

Comment:
so for my understanding, is this map reflecting the desired state of a unique id ( this case the onboardUrl from spec ) and then you get the existing configuration after and put it in another map with same format? Those two maps are then passed to the manage_state function which decides what to delete / add / update?

what is the purpose of the normalized_payload and why is this not done for existing? are you conforming the input to the existing configuration?